### ISSUE TITLE
xcodebuild exited with code '65' and signal 'null'

### ISSUE THREAD

**Issue by RZahr** [appium/appium#20826]
xcodebuild exited with code '65' and signal 'null'

Anyone is experiencing this ? I saw old issues about this but i didnt see a proper step by step guide on how to solve it. on Android i only need to install the driver and i am set but on iOS it seems not enough. I tried updating to the latest driver but still same issues apply. I have xcode Version 16.0 (16A242d). In case any more info is need i will gladly share them.

**Comment by mykola-mokhnach** [comment]
Please provide a proper issue description with the requested information

**Comment by RZahr** [comment]
> Please provide a proper issue description with the requested information

I updated the description with extra info. Please let me know if i should add anything else to it

**Comment by mykola-mokhnach** [comment]
Yes, we also need the full server log with xcodebuild logging enabled.

**Comment by mykola-mokhnach** [comment]
>     "appium:usePrebuiltWDA": true,

> [09e4c943][Xcode]     /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild test-without-building -project /Users/xx/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/xx/.appium/node_modules/appium-xcuitest-driver/node_modules -destination id=6EB082CB-2590-45FB-8A6B-89AC7890E0C1 IPHONEOS_DEPLOYMENT_TARGET=18.0 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO


The above error is expected. Capabilities expect WDA to be already built but it is not.

**Comment by RZahr** [comment]
> > ```
> > "appium:usePrebuiltWDA": true,
> > ```
> 
> > [09e4c943][Xcode]     /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild test-without-building -project /Users/xx/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/xx/.appium/node_modules/appium-xcuitest-driver/node_modules -destination id=6EB082CB-2590-45FB-8A6B-89AC7890E0C1 IPHONEOS_DEPLOYMENT_TARGET=18.0 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO
> 
> The above error is expected. Capabilities expect WDA to be already built but it is not.

Even with usePrebuiltWDA as false i still get an error. I will re-update the above server log having it false. i am running it now

**Comment by RZahr** [comment]
> > ```
> > "appium:usePrebuiltWDA": true,
> > ```
> 
> > [09e4c943][Xcode]     /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild test-without-building -project /Users/xx/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/xx/.appium/node_modules/appium-xcuitest-driver/node_modules -destination id=6EB082CB-2590-45FB-8A6B-89AC7890E0C1 IPHONEOS_DEPLOYMENT_TARGET=18.0 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO
> 
> The above error is expected. Capabilities expect WDA to be already built but it is not.

yeah i tried it with usePrebuiltWDA as false and now it is working. But randomly as mentioned it fails. I will keep an eye on it today and in case of any failure i will add another comment with the server logs
